src/libsystemd-network/sd-dhcp6-client.c

```diff
@@ -1319,17 +1362,38 @@ static int client_receive_message(
 
         case DHCP6_STATE_BOUND:
         case DHCP6_STATE_STOPPED:
+        case DHCP6_STATE_STOPPING:
         default:
                 assert_not_reached();
         }
 
         return 0;
 }
 
+static void client_send_release(sd_dhcp6_client *client) {
+        int r;
+        sd_dhcp6_lease *lease;
+
+        if (sd_dhcp6_client_get_lease(client, &lease) < 0)
+                return;
+
+        if (!lease->ia_na && !lease->ia_pd)
+                return;
+
+        client_set_state(client, DHCP6_STATE_STOPPING);
+        r = dhcp6_client_send_message(client);
+        if (r < 0)
+                log_dhcp6_client_errno(client, r,
+                                       "Failed to send to DHCP6 release message: %m");
```

Comment:
Please propagate the failure to the caller, and ignore the failure in the caller side:
```c
static int client_send_release(sd_dhcp6_client *client) {
        ...;
        if (r < 0)
                return log_dhcp6_client_errno(client, r, "...");

        return 0;
}

int sd_dhcp6_client_stop(sd_dhcp6_client *client) {
        ...;

        (void) client_send_release(client);

        ...;
}
```